ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user-settings/user-settings.tsx

```diff
@@ -107,6 +107,25 @@ class UserSettings extends React.Component<Props, State> {
   render() {
     const { extensions } = this.props
     const { component } = this.state
+    const extensionComponents = extensions.map((extension: SettingsComponent) => {
+      if(extension.custom) {
```

Comment:
consider a design avoids this `extension.custom` flag

Reply:
looks like this is the only place `extension.children` is now used. Could you just use if(extension.children) instead?

Reply:
sounds good to me - maybe rename `extension.children` to `extension.customContent` too

Reply:
tbh I still find it confusing that you can define a SettingsComponent that has a text, icon, and customContent and the first two members will be ignored completely

Reply:
I'm not a huge fan of it either to be completely honest but I can't think of another way either.

The original idea here was to just make all the extensions react components that get passed in, where all the components have complete control over their lifecycle events, click handlers, etc. But that doesn't work because of L121.

Reply:
maybe split `SettingsComponent` into two types and accept either. or some solution involving interfaces/classes

Reply:
That seems reasonable, I can do something like `SettingsComponent` and `CustomSettingsComponent`, and do something like this: https://medium.com/ovrsea/checking-the-type-of-an-object-in-typescript-the-type-guards-24d98d9119b0
to find out which type each component is.

Reply:
type-checking code is awkward - maybe you don't need it? https://github.com/zta6/ddf/compare/DDF-5102...rymach:pr/5103

Reply:
played with that a bit, doesn't quite seem to work. Tried a couple variations of that and keep getting 
```
      TS2345: Argument of type '({ component: Element; text: string; icon: string; } | { component: Element; })[]' is not assignable to parameter of type 'SettingsComponent[]'.
  Type '{ component: Element; text: string; icon: string; } | { component: Element; }' is not assignable to type 'SettingsComponent'.
    Type '{ component: Element; }' is not assignable to type 'SettingsComponent'.
      Type '{ component: Element; }' is not assignable to type 'SimpleSettingsComponent'.
        Property 'text' is missing in type '{ component: Element; }'.
```
coming from the line where the extensions are passed in the downstream project